File: sample/objective-c/GNAdSampleAdMobAdapter/GNAdSampleAdMobAdapter/ViewController.m
//
//  ViewController.m
//  GNAdSampleAdMobAdapter
//

@import GoogleMobileAds;

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    // Create a view of the standard size at the top of the screen.
    _bannerView = [[GADBannerView alloc] initWithFrame:CGRectMake(0.0,20.0,
                                                                  GAD_SIZE_320x50.width,
                                                                  GAD_SIZE_320x50.height)];
    
    // Change the ad unit ID to AdMob_Mediation_ID.
    _bannerView.adUnitID = @"YOUR_ADMOB_BANNER_UNIT_ID";
    _bannerView.delegate = self;
    _bannerView.rootViewController = self;
    [self.view addSubview:_bannerView];
    
    _bannerView.center = CGPointMake(self.view.center.x, _bannerView.center.y);
    // Load the ads with a general ad request.
    [_bannerView loadRequest:[GADRequest request]];
}

#pragma mark GADBannerViewDelegate impl

// We've received an ad successfully.
- (void)adViewDidReceiveAd:(GADBannerView *)adView {
    NSLog(@"Received ad successfully");
}

- (void)adView:(GADBannerView *)view
didFailToReceiveAdWithError:(GADRequestError *)error {
    NSLog(@"Failed to receive ad with error: %@", [error localizedFailureReason]);
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: sample/use_cocoapods/objectivec/GNAdDFPBannerAdapterSample/GNAdDFPBannerAdapterSample/ViewController.swift
//
//  ViewController.swift
//  GNAdDFPBannerAdapterSample
//
//  Created by Geniee on 2018/07/25.
//  Copyright © 2018年 Geniee. All rights reserved.
//

import UIKit
import GoogleMobileAds

class ViewController: UIViewController {
    var bannerView: DFPBannerView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        bannerView = DFPBannerView(adSize: kGADAdSizeMediumRectangle)
        bannerView.adUnitID = "MY_DFP_OR_ADMOB_AD_UNIT_ID"
        bannerView.rootViewController = self
        bannerView.load(DFPRequest())

        addBannerViewToView(bannerView)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    /// Tells the delegate an ad request loaded an ad.
    func adViewDidReceiveAd(_ bannerView: DFPBannerView) {
        print("adViewDidReceiveAd")
    }
    
    /// Tells the delegate an ad request failed.
    func adView(_ bannerView: DFPBannerView,
                didFailToReceiveAdWithError error: GADRequestError) {
        print("adView:didFailToReceiveAdWithError: \(error.localizedDescription)")
    }
    
    func addBannerViewToView(_ bannerView: UIView) {
        bannerView.translatesAutoresizingMaskIntoConstraints = false
        view.addSubview(bannerView)
        if #available(iOS 11.0, *) {
            positionBannerAtBottomOfSafeArea(bannerView)
        }
        else {
            positionBannerAtBottomOfView(bannerView)
        }
    }
    
    @available (iOS 11, *)
    func positionBannerAtBottomOfSafeArea(_ bannerView: UIView) {
        // Position the banner. Stick it to the bottom of the Safe Area.
        // Centered horizontally.
        let guide: UILayoutGuide = view.safeAreaLayoutGuide
        
        NSLayoutConstraint.activate(
            [bannerView.centerXAnchor.constraint(equalTo: guide.centerXAnchor),
             bannerView.centerYAnchor.constraint(equalTo: guide.centerYAnchor)]
        )
    }
    
    func positionBannerAtBottomOfView(_ bannerView: UIView) {
        // Center the banner horizontally.
        view.addConstraint(NSLayoutConstraint(item: bannerView,
                                              attribute: .centerX,
                                              relatedBy: .equal,
                                              toItem: view,
                                              attribute: .centerX,
                                              multiplier: 1,
                                              constant: 0))
        view.addConstraint(NSLayoutConstraint(item: bannerView,
                                              attribute: .centerY,
                                              relatedBy: .equal,
                                              toItem: view,
                                              attribute: .centerY,
                                              multiplier: 1,
                                              constant: 0))
    }


}

